src/SFML/Network/Http.cpp

```diff
@@ -261,7 +261,10 @@ void Http::Response::parse(const std::string& data)
             std::istreambuf_iterator<char> it(in);
             std::istreambuf_iterator<char> itEnd;
             for (std::size_t i = 0; ((i < length) && (it != itEnd)); i++)
-                m_body.push_back(*it++);
+            {
+                m_body.push_back(*it);
+                ++it;
```

Comment:
Thanks for the awesome Godbolt link! Because subsequent readers may not pick up on why we're doing this, can you add a quick comment mentioning how this is a GCC 12.1 workaround? Maybe something like this?

```suggestion
                ++it; // Iterate in separate expression to work around false positive -Wnull-dereference warning in GCC 12.1.0
```

Reply:
Not a problem! Added.